src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -90,18 +128,28 @@ def get_source_schema(cls):
         source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file."
         return source_schema
 
-    def __init__(self, file_path: FilePathType, stub_test: Optional[bool] = False):
-        super().__init__(file_path=str(file_path))
+    def __init__(
+        self,
+        file_path: FilePathType,
+        stub_test: Optional[bool] = False,
+        spikeextractors_backend: Optional[bool] = False,
+    ):
+        if spikeextractors_backend:
+            self.RX = se.SpikeGLXRecordingExtractor
+            super().__init__(file_path=str(file_path))
+            self.recording_extractor = OldToNewRecording(oldapi_recording_extractor=self.recording_extractor)
+        else:
+            file_path = Path(file_path)
+            folder_path = file_path.parent
+            stream_id = "".join(file_path.suffixes[:-1])[1:]
```

Comment:
How robust is this parsing sub-function for the stream id? What if the stem has been altered slightly, say to include extra '.'?

Reply:
Yes, I saw that and it bothers me. I added now an extra utility to do the parsing in a more robust and, in my view, more transparent / explicit fashion.